### PR TITLE
chore: Add dependabot config

### PR DESCRIPTION
**Pull request by ActuallyACat**
# Objective
This config will supercede the org-level defaults for Dependabot. 

**If you're new to Kaizen, please ask #prod_design_systems to set up an onboarding session to get you up to speed.** If you have an urgent PR to merge before that happens, it is safest to ask in #prod_design_systems to have a design system advocate review the PR to catch any issues.

### CODE REVIEW

**Comment by mbylstra** [comment]
Ah, so this is going back to the old dependabot V1 where it should only get updates on the weekend? Was that the intention?

**Comment by mbylstra** [comment]
I noticed that v1 is being shut down on Aug 1: https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/upgrading-from-dependabotcom-to-github-native-dependabot

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+version: 1
```
Comment (mbylstra) [inline .github/dependabot.yml:1]: is version 1 intentional? Not version 2?

Comment (ActuallyACat) [reply .github/dependabot.yml]: Oops - fixed

**Comment by ActuallyACat** [comment]
> Ah, so this is going back to the old dependabot V1 where it should only get updates on the weekend? Was that the intention?

My bad! I meant to use v2 - all other parts in the config relate to v2 config. There are some things the v2 config can do that v1 can't - which is limit the number of total pull requests and ignore patches. I will probably need to tweak it some more depending on how noisy / quiet it is but I'm hoping this is a good start.

**Review by mbylstra — APPROVED** [review]

**Pull request merged**